### PR TITLE
Clean up ExperimentalMutableObjectManager

### PR DESCRIPTION
**Pull request by jackhumphries**
## Why are these changes needed?

This PR improves the coding style of ExperimentalMutableObjectManager.

## Related issue number

N/A

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/core_worker/experimental_mutable_object_manager.cc [hunk 1]
```diff
@@ -159,230 +172,191 @@ void ExperimentalMutableObjectManager::DestroySemaphores(const ObjectID &object_
...
+                                          int64_t data_size,
+                                          const uint8_t *metadata,
+                                          int64_t metadata_size,
+                                          int64_t num_readers,
+                                          std::shared_ptr<Buffer> &data) {
+  WriterChannel *channel = GetChannel<WriterChannel>(writer_channels_, object_id);
+  if (!channel) {
     return Status::ObjectNotFound("Writer channel has not been registered");
   }
+  RAY_CHECK(!channel->written);
```
Comment (stephanie-wang) [inline src/ray/core_worker/experimental_mutable_object_manager.cc:185]: What's the reasoning for changing this from returning a status to asserting?

File: src/ray/core_worker/experimental_mutable_object_manager.cc [hunk 2]
```diff
@@ -159,230 +172,191 @@ void ExperimentalMutableObjectManager::DestroySemaphores(const ObjectID &object_
...
+  ReaderChannel *channel = GetChannel<ReaderChannel>(reader_channels_, object_id);
+  if (!channel) {
     return Status::ObjectNotFound("Reader channel has not been registered");
   }
-  auto &channel = reader_channel_entry->second;
 
-  // This lock ensures that there is only one reader at a time. The lock is released in
-  // `ReadRelease()`.
-  channel.lock->Lock();
+  channel->lock->Lock();
```
Comment (stephanie-wang) [inline src/ray/core_worker/experimental_mutable_object_manager.cc:232]: Could you add back the comment that was here?

File: src/ray/core_worker/experimental_mutable_object_manager.cc [hunk 3]
```diff
@@ -159,230 +172,191 @@ void ExperimentalMutableObjectManager::DestroySemaphores(const ObjectID &object_
...
-      channel.mutable_object->header->ReadRelease(sem, channel.next_version_to_read));
+      channel->mutable_object->header->ReadRelease(sem, channel->next_version_to_read));
   // The next read needs to read at least this version.
-  channel.next_version_to_read++;
-  // This lock ensures that there is only one reader at a time. The lock is acquired in
-  // `ReadAcquire()`.
-  channel.lock->Unlock();
+  channel->next_version_to_read++;
 
+  channel->lock->Unlock();
```
Comment (stephanie-wang) [inline src/ray/core_worker/experimental_mutable_object_manager.cc:271]: Same here, could you add back the comment?

**Review by stephanie-wang — COMMENTED** [review]
Thanks, just a few small comments!

**Review by stephanie-wang — APPROVED** [review]

File: src/ray/core_worker/experimental_mutable_object_manager.cc (hunk 1 above)
Comment (jackhumphries) [reply src/ray/core_worker/experimental_mutable_object_manager.cc:185]: Discussed offline. Done.

File: src/ray/core_worker/experimental_mutable_object_manager.cc (hunk 2 above)
Comment (jackhumphries) [reply src/ray/core_worker/experimental_mutable_object_manager.cc:232]: Done.

File: src/ray/core_worker/experimental_mutable_object_manager.cc (hunk 3 above)
Comment (jackhumphries) [reply src/ray/core_worker/experimental_mutable_object_manager.cc:274]: Done.

**Pull request merged**